isis/tests/FileNameTests.cpp

```diff
@@ -111,47 +124,117 @@ TEST(FileName, isQuestionMarksNoExtensionVersioned) {
   QString test = "/testy/mc/test/face/test??????";
   FileName file(test);
   
-  EXPECT_EQ(true, file.isVersioned());
+  EXPECT_TRUE(file.isVersioned());
 }
 
 TEST(FileName, isQuestionMarksExtensionVersioned) {
   QString test = "/testy/mc/test/face/test??????.cub";
   FileName file(test);
   
-  EXPECT_EQ(true, file.isVersioned());
+  EXPECT_TRUE(file.isVersioned());
 }
 
 TEST(FileName, isDDMMMYYYVersioned) {
   QString test = "/testy/mc/test/face/test{ddMMMyyyy}..cub";
   FileName file(test);
   
-  EXPECT_EQ(true, file.isVersioned());
+  EXPECT_TRUE(file.isVersioned());
 }
 
-//TODO Waiting for GMock
+TEST(FileName, HighestVersion) {
+  std::ofstream fileStream;
+  fileStream.open("test000001.cub");
+  fileStream << "test";
+  fileStream.close();
+  fileStream.open("test000002.cub");
+  fileStream << "test";
+  fileStream.close();
+  
+  FileName file("test??????.cub");
+  EXPECT_EQ("./test000002.cub", file.highestVersion().expanded());
+  remove("test000001.cub");
+  remove("test000002.cub");
+  
+}
```

Comment:
If you run into an exception during this test, the files won't get cleaned up. You can use a fixture's setup and teardown methods to setup the files and ensure that they get removed even if the test doesn't run to completion.